### ISSUE TITLE
Revise build flags and instructions necessary for cross-compilation

### ISSUE THREAD

**Issue by jpy-luke** [menu-hanke/m2#2]
When cross-compiling for Windows, using mingw, on a Linux host
`make CROSS=x86_64-w64-mingw32- TARGET_SYS=Windows dependencies`
`make CC=x86_64-w64-mingw32-gcc OS=Windows XBCLFLAGS="-o windows" DEPS_LUAJIT=/usr/bin/luajit`
will run succesfully.

Possibly the OS flag has been geared for the situation where the build host is Windows. It will direct the build process to use luajit.exe as the local luajit executable. This obviously won't work on a Linux host. DEPS_LUAJIT can be used to circumvent this, by using another available luajit executable.

Additionally the usage of the XBCLFLAGS to direct the bcloader system remains undocumented.

README.md needs to address the cross-compilation situation.
The build system should be further parameterized to take into account the build host OS in addition to the target OS (see the LuaJIT package build for reference).

**Comment by vfprintf** [comment]
Agree. The current Makefile is written for local builds. I think I'll put those options behind a `TARGET_SYS` flag, like luajit does. The host OS doesn't really matter, except for some default paths (`DEPS_LUAJIT`), since we don't build any code to run on the host.
